verifier/rest/rest.go

```diff
@@ -230,20 +237,67 @@ func convertRequestToREST(request verifier.VerifyAttestationRequest) *ccpb.Verif
 		TokenOptions: convertTokenOptionsToREST(request.TokenOptions),
 	}
 
-	if request.Attestation.GetSevSnpAttestation() != nil {
-		sevsnp, err := convertSEVSNPProtoToREST(request.Attestation.GetSevSnpAttestation())
-		if err != nil {
-			log.Fatalf("Failed to convert SEVSNP proto to API proto: %v", err)
+	if request.Attestation != nil {
+		// TPM attestation route
+		quotes := make([]*confidentialcomputingpb.TpmAttestation_Quote, len(request.Attestation.GetQuotes()))
+		for i, quote := range request.Attestation.GetQuotes() {
+			pcrVals := map[int32][]byte{}
+			for idx, val := range quote.GetPcrs().GetPcrs() {
+				pcrVals[int32(idx)] = val
+			}
+
+			quotes[i] = &confidentialcomputingpb.TpmAttestation_Quote{
+				RawQuote:     quote.GetQuote(),
+				RawSignature: quote.GetRawSig(),
+				HashAlgo:     int32(quote.GetPcrs().GetHash()),
+				PcrValues:    pcrVals,
+			}
 		}
-		verifyReq.TeeAttestation = sevsnp
-	}
 
-	if request.Attestation.GetTdxAttestation() != nil {
-		tdx, err := convertTDXProtoToREST(request.Attestation.GetTdxAttestation())
-		if err != nil {
-			log.Fatalf("Failed to convert TD quote proto to API proto: %v", err)
+		certs := make([][]byte, len(request.Attestation.GetIntermediateCerts()))
+		for i, cert := range request.Attestation.GetIntermediateCerts() {
+			certs[i] = cert
+		}
+
+		verifyReq.TpmAttestation = &confidentialcomputingpb.TpmAttestation{
+			Quotes:            quotes,
+			TcgEventLog:       request.Attestation.GetEventLog(),
+			CanonicalEventLog: request.Attestation.GetCanonicalEventLog(),
+			AkCert:            request.Attestation.GetAkCert(),
+			CertChain:         certs,
+		}
+
+		if request.Attestation.GetSevSnpAttestation() != nil {
```

Comment:
Why are we handling this? If this is for the cmdline, let's go and make those changes to force the cmdline to use the API correctly rather than have this complex logic

Reply:
The logic is added here https://github.com/google/go-tpm-tools/commit/b30d3733c5835cb35cfeabc68a194bbf98f18fd7
I think it's for the cmd tool.

I think we currently don't have a seperate API for SNP yet.

We probably need to create seperate PR for this.